### PR TITLE
Testing To Do list: part 1

### PR DESCRIPTION
**Pull request by ridwanediallo**
# Project requirements:

1️⃣ This is a pair-programming project. Please decide whose project will you use for testing.

2️⃣ Use jest framework for testing.

3️⃣ Create a test file ([..].test.js) for a file containing the add item and delete item functions that you want to test.

4️⃣ Take a look at each of these functions. Are they pure functions? If the answer is "yes" then writing tests for them should be 
 straightforward. Some of those functions however will update localStorage and manipulate the DOM. For those, you will need to use mocks:

- Mock a storage object to "imitate" localStorage operations (only if you are using jest < 24.x, in newer versions localStorage is mocked automatically).

- Mock HTML to test if add/delete functions add or remove exactly one <li> element to/from the list in the DOM.

5️⃣ Make sure you group your tests using the describe() method.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Additional description about the project and its features.
 
 To get a local copy up and running follow these simple steps.
```
Comment (SajjadAhmad14) [inline README.md:16]: Please add `npm install`, `npm start` as these commands are necessary to set the project up locally. Moreover, Please add the `npm test` command so the user is able to run the tests. Kindly make sure the README is professional and has all the information for the end user.

File: src/__tests__/taskList.test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+const todo = new TaskList();
+
+describe('Testing add functionality', () => {
+  test('Add todo', () => {
+    todo.addTodo('Have fun', todo.tasks);
+    expect(todo.tasks[0].desp).toEqual('Have fun');
+  });
+});
+
```
Comment (SajjadAhmad14) [inline src/__tests__/taskList.test.js:15]: Please test the functions `#clearCompleted` and `#clickCheck` and use mock for functions that use `localStorage`. Kindly make sure all the methods for adding and removing the items are tested.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi team, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and description. :heavy_check_mark:
All tests are passed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: README.md (hunk 1 above)
Comment (ridwanediallo) [reply README.md:16]: Thank you for taking the time to review the project. However, I don't understand what you mean by this requirement. 
Do you want me to add npm start, npm install, and npm test in the readme ? 
Could you be more specific?

File: src/__tests__/taskList.test.js (hunk 2 above)
Comment (ridwanediallo) [inline src/__tests__/taskList.test.js:15]: Sorry, but from my understanding, these are not among the requirement for part one of this project.
Please find the requirement link here: https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/tests_1.md

**Review by er-contreras — APPROVED** [review]
Hi @ridwanediallo , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

- I see that you did most of the work, well done!

Felix

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**